packages/amplify_auth_cognito/ios/Classes/FlutterUpdateUserAttributeResult.swift

```diff
@@ -0,0 +1,116 @@
+
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import Foundation
+import Amplify
+
+struct FlutterUpdateUserAttributeResult {
+    var isUpdated: Bool
+    var updateAttributeStep: String;
+    var additionalInfo: [String: String]
+    var codeDeliveryDetails: [String: String]
+
+    init(res: AmplifyOperation<AuthUpdateUserAttributeRequest, AuthUpdateAttributeResult, AuthError>.OperationResult){
+      self.isUpdated = isComplete(res: res)
+      self.updateAttributeStep = setState(res: res)
+      self.additionalInfo = setAdditionalInfo(res: res)
+      self.codeDeliveryDetails = setCodeDeliveryDetails(res: res)
+    }
+
+    func toJSON() -> Dictionary<String, Any> {
+      return [
+        "isUpdated": self.isUpdated,
+        "nextStep": [
+            "updateAttributeStep": self.updateAttributeStep,
+            "additionalInfo": self.additionalInfo,
+            "codeDeliveryDetails": self.codeDeliveryDetails
+        ]
+      ]
+    }
+}
+
+func isComplete(res: AmplifyOperation<AuthUpdateUserAttributeRequest, AuthUpdateAttributeResult, AuthError>.OperationResult) -> Bool {
```

Comment:
*Note:* Several of these util methods are duplicated throughout a few classes. I followed the pattern from the other classes and duplicated them here, but I think they should be broken out.